toolkit/scripts/analysis.mk

```diff
@@ -86,3 +89,49 @@ sodiff-check: $(BUILT_PACKAGES_FILE) | $(SODIFF_REPO_FILE)
 	<$(BUILT_PACKAGES_FILE) $(SODIFF_SCRIPT) $(RPMS_DIR)/ $(SODIFF_REPO_FILE) $(RELEASE_MAJOR_ID) $(SODIFF_OUTPUT_FOLDER)
 
 package-toolkit: $(SODIFF_REPO_FILE)
+
+######## LICENSE CHECK ########
+
+license_check_build_dir   = $(BUILD_DIR)/license_check_tool
+license_out_dir           = $(OUT_DIR)/license_check
+license_results_file_pkg  = $(license_out_dir)/license_check_results_pkg.json
+license_results_file_imge = $(license_out_dir)/license_check_results_image_$(config_name).json
+
+.PHONY: license-check license-check-img clean-license-check
+
+clean: clean-license-check
+clean-license-check:
+	@echo Verifying no mountpoints present in $(license_check_build_dir)
+	$(SCRIPTS_DIR)/safeunmount.sh "$(license_check_build_dir)" && \
+	rm -rf $(license_check_build_dir) && \
+	rm -rf $(license_out_dir)
+
+# If we are using the default RPMS_DIR as LICENSE_CHECK_DIRS, we are responsible for building the rpms as needed.
+# Can set REBUILD_PACKAGES=n to skip rebuilding the rpms if they are already built.
+ifeq ($(LICENSE_CHECK_DIRS),$(RPMS_DIR))
+license-check: $(LICENSE_CHECK_DIRS)
+endif
```

Comment:
I'm not sure about this part, because it by default ties the licensing tool to the package building functionality of the toolkit. And it's conditional in an obscure way for the user - they change `LICENSE_CHECK_DIRS` and suddenly you see different behaviour. If anything, I'd go to the other way around: conditionally add license checking to package building but let the licensing tool be its own, independent component.

Reply:
I think it's important to be consistent, and our other tools can automatically call dependencies when it makes sense. I'll add a 3rd target that doesn't do so if that is a sticking point.